pkgs/by-name/ka/kanidm/generic.nix

```diff
@@ -35,38 +35,37 @@
 let
   arch = if stdenv.hostPlatform.isx86_64 then "x86_64" else "generic";
 
-  versionUnderscored = builtins.replaceStrings [ "." ] [ "_" ] (
-    lib.versions.majorMinor kanidm.version
-  );
-
-  provisionPatches = [
-    (./. + "/provision-patches/${versionUnderscored}/oauth2-basic-secret-modify.patch")
-    (./. + "/provision-patches/${versionUnderscored}/recover-account.patch")
-  ];
-
   upgradeNote = ''
     Please upgrade by verifying `kanidmd domain upgrade-check` and choosing the
     next version with `services.kanidm.package = pkgs.kanidm_1_x;`
 
     See upgrade guide at https://kanidm.github.io/kanidm/master/server_updates.html
   '';
+
+  versionUnderscored =
+    finalAttrs: lib.replaceStrings [ "." ] [ "_" ] (lib.versions.majorMinor finalAttrs.version);
```

Comment:
Your PR description mentioned not using finalAttrs here, so I am a bit surprised to see this using finalAttrs.version. Am I missing something or is this a bug?

(As an aside, I'm fine with the one-time cost of moving the let block inside the finalAttrs function. PR review has the option to hide whitespace changes which makes the review itself easier)

Reply:
You made me realize that I moved `versionUnderscored` from above `upgradeNote` to below that for now real reason. I amended that commit to drop the unnecessary ordering differences and new newlines.

I would still prefer to keep the change as is, with `versionUnderscored` now being a function, instead of moving the `let in` in the same commit. But I can add another commit that does just that, if you want. Or, you know, you say that you, as the maintainer, want it to be in the same commit, in which case I would do just that too :)

Reply:
My opinion is definitely not that strong here.

`v` does come after `u`, so maybe that's why you moved it? :) Either way, this looks good to me.